src/mon/MgrMonitor.cc

```diff
@@ -647,6 +649,10 @@ bool MgrMonitor::prepare_beacon(MonOpRequestRef op)
     dout(10) << "no change" << dendl;
   }
 
+  if (plugged) {
```

Comment:
Do we need a local variable to keep track of  whether paxos is plugged?
Can't we do this instead,
```
if (paxos.isplugged()){
    paxos.unplug();
}
```

Likewise in changes to `MgrMonitor::tick()` and `MgrMonitor::prepare_command()`

Reply:
The point here is that we only unplug paxos if `drop_active` actually plugged it. Like the OSDMonitor's blocklist method, sometimes PaxosServices call into each other. For this reason, I think it's appropriate to not unconditionally unplug paxos.